Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1344,7 +1344,7 @@ ExceptionOr<void> HTMLElement::showPopover()
         document().hideAllPopoversUntil(ancestor.get(), FocusPreviousElement::No, FireEvents::Yes);
 
         if (popoverState() != originalState)
-            return { };
+            return Exception { InvalidStateError, "The value of the popover attribute was changed while hiding the popover."_s };
```

Comment:
I don't see this exception in the spec: https://html.spec.whatwg.org/#show-popover (Step 7.5)

@annevk Should this be added to the spec (in which case we should file an issue) or should the WPT be updated to match the spec?

Reply:
cc @josepharhar @mfreed7 in case they have opinions about this particular bit.

Reply:
(This exception seems OK to add IMO, though I just want to make sure everything gets put in the spec)

Reply:
I did read 7.5 too quickly. I think things depend on the behaviour of throwExceptions being true and https://html.spec.whatwg.org/#check-popover-validity throwing an Exception. If the expectation is that the show popover algorithm passes that exception on (for throwExceptions being true) then everything is covered.

Reply:
Anne pointed to this which I think resolves this discussion:
https://infra.spec.whatwg.org/#algorithm-control-flow

Reply:
The way I read the algorithm doesn't require throwing the exception, since it's an "or", so we should be able to bail early if one of the conditions matches. So the popover validity check can be separate (which is how it currently is before this change).

Now if the "or" requires to evaluate both conditions, then we should probably move the checkPopoverValidity check up before the `if (popoverState() != originalState)` check. That would require no spec change.

Reply:
@nt1m I think that's correct. What I was clarifying to @rwlbuis is that algorithms rethrow, but in this case you don't hit the algorithm that throws.

Reply:
I filed https://github.com/whatwg/html/issues/9037 for this.

Reply:
Yeah I'm not sure what the motivation was for only checking for popover attribute state changes (between manual and auto) in show popover and not hide popover. It may have been intentional though